docs/decisions/0018-units-as-containers.rst

```diff
@@ -0,0 +1,78 @@
+18. Modeling Units as a Concrete Implementation of the Container Capability
```

Comment:
I'm not sure if this deserves its own ADR, but it helps illustrate the decisions using a more familiar concept like units.